### ISSUE TITLE
[Bug][Core] No args to ray.method() leads to confusing failure

### ISSUE THREAD

**Issue by DmitriGekhtman** [ray-project/ray#23271]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### Issue Severity

Low: It annoys or frustrates me for a moment.

### What happened + What you expected to happen

New users (apparently me when I wrote this [test](https://github.com/ray-project/ray/blob/814b49356c1e773f2ec8d7c643acef48b6db08cf/python/ray/tests/kubernetes_e2e/test_k8s_operator_basic.py#L302)) may think that `ray.method` is required to decorate actor methods.
(The purpose of `ray.method` is to pass task options to a method of an actor.)

Three issues: 
(1) The naming and purpose of `ray.method` is obscure.
(2) Using ray.method with no arguments fails. Running this code
```python
import ray

@ray.remote
class A:
    @ray.method()
    def f(self):
        return
```
yields an uninterpretable exception
```
Traceback (most recent call last):
  File "test2.py", line 5, in <module>
    @ray.remote
  File "test2.py", line 7, in Test
    @ray.method()
  File "/Users/dmitrigekhtman/anaconda3/envs/anyscale7/lib/python3.7/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/Users/dmitrigekhtman/anaconda3/envs/anyscale7/lib/python3.7/site-packages/ray/actor.py", line 55, in method
    assert len(kwargs) == 1
AssertionError
```
(3) For Ray <=1.11.0 running the above code while connected to Ray client does not yield an exception.
In other words,
`ray start --head`
```python
import ray

ray.init("ray://127.0.0.1:10001")
@ray.remote
class A:
    @ray.method()
    def f(self):
        return
```
is fine!
However, running the last snippet **does yield an exception in Ray 1.12.0rc0**. Thus, this could be viewed as a (minor) regression. It has a low chance of breaking someone's code.

### Versions / Dependencies

See above.

### Reproduction script

See above

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by DmitriGekhtman** [comment]
cc core oncall @stephanie-wang 
@ckw017 for the Ray client piece
@naijoaix for 1.12.0 release management

**Comment by stephanie-wang** [comment]
Hmm I don't think this is a valid release blocker. We've always had the same behavior for @ray.remote.

**Comment by DmitriGekhtman** [comment]
@stephanie-wang there is a change in the behavior of @ray.method within a Ray client connection but the impact is trivial

**Comment by DmitriGekhtman** [comment]
Impact is sufficiently trivial to remove from release blockers.

**Comment by DmitriGekhtman** [comment]
There's a theme of Ray's APIs failing to validate user input parameters.
https://github.com/ray-project/ray/issues/23166
Kind of a Python problem.

**Comment by ckw017** [comment]
Looks like the client piece is from an update to ray.method #22810, but IMO we're probably better off leaving in this behavior so that client mode is consistent with non-client

**Comment by DmitriGekhtman** [comment]
It'd be nice if both gave the user a helpful error message.
